src/SimMatrix.cpp

```diff
@@ -309,3 +309,35 @@ SimMatrix SimMatrix::DoSimStepReturnMatrix(const MatrixSetup& setup)
 	*this = locSimMatrix;
 	return locSimMatrix;
 }
+
+// only for cells not from edges (with margins)
+void SimMatrix::MyLifeNeighbors(SimMatrix& sMatrix)
+{
+    for (auto i = sMatrix.simMatrix.begin() + 1; i != sMatrix.simMatrix.end() - 1; ++i)
+	{
+		for (auto j = i->begin() + 1; j != i->end() - 1; ++j)
+		{
+            // around the cell
+            for (int xPosRelative = -1; xPosRelative < 2; xPosRelative++)
+                for ( int yPosRelative = -1; yPosRelative < 2; yPosRelative++)
+                    if ((i + xPosRelative)->begin() + yPosRelative == j)
+                        // if himself do nothing
+                        continue;
+                    else if (((i + xPosRelative)->begin() + yPosRelative )->isAlive())
+                        ((i + xPosRelative)->begin() + yPosRelative )->IncreaseMyLifeNeighbors();;
+                    else
+                        continue;
+		}
+	}
+}
+
+void SimMatrix::ResetLifeNeighborsCounterAll(SimMatrix& sMatrix)
```

Comment:
Is not called anywhere.

Reply:
Yup, this prepare for next refactories code step when this function will be use